python/foglamp/services/core/api/filters.py

```diff
@@ -0,0 +1,270 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+import json
+import copy
+from aiohttp import web
+from foglamp.common.storage_client.payload_builder import PayloadBuilder
+from foglamp.common.configuration_manager import ConfigurationManager
+from foglamp.services.core import connect
+from foglamp.common.storage_client.exceptions import StorageServerError
+from foglamp.services.core.api import utils as apiutils
+from foglamp.common import logger
+
+__author__ = "Massimiliano Pinto"
+__copyright__ = "Copyright (c) 2018 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_help = """
+    ---------------------------------------------------------------------------
+    | POST            | /foglamp/filter                                       |
+    | PUT             | /foglamp/filter/{service_name}/pipeline               |
+    ---------------------------------------------------------------------------
+"""
+
+_LOGGER = logger.setup("filter")
+
+
+async def create_filter(request):
+    """
+    Create a new filter with a specific plugin
+    
+    :Example:
+     curl -X POST http://localhost:8081/foglamp/filter -d 
+     '{
+        "name": "North_Readings_to_PI_scale_stage_1Filter",
+        "plugin": "scale"
+     }'
+
+    'name' is the filter name
+    'plugin' is the filter plugin name
+
+    The plugin is loaded and default config from 'plugin_info'
+    is fetched.
+
+    A new config category 'name' is created:
+    items are:
+       - 'plugin'
+       - all items from default plugin config
+
+    NOTE: The 'create_category' call is made with keep_original_items = True
+
+    """
+    try:
+        # Get inpout data
+        data = await request.json()
+        # Get filter name
+        filter_name = data.get('name', None)
+        # Get plugin name
+        plugin_name = data.get('plugin', None)
+
+        # Check we have needed input data
+        if not filter_name or not plugin_name:
+            raise web.HTTPBadRequest(reason='Filter name or plugin name are required.')
```

Comment:
`return web.HTTPBadRequest(reason='Filter name or plugin name are required.')`. Otherwise it will return 500 HTTP status code with same message. We need 400 for this case.

Reply:
fixed